### PR TITLE
DB-9612 pass result column data types of ProjectRestrictOperation dow…

### PR DESCRIPTION
**Pull request by yxia92**
…n explicitly to execution

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1230/

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/DataTypeDescriptor.java [hunk 1]
```diff
@@ -1752,5 +1753,20 @@ public void setChildren(DataTypeDescriptor[] children) {
     public DataValueDescriptor getDefault() throws StandardException {
         return typeId.getDefault();
     }
+
+    public StructField getStructField(String columnName) {
+        StructField childStructField = null;
+        if (typeId.isArray()) {
+            // get the child type descriptor & struct field
+            assert typeDescriptor!=null:"Catalog Type";
+            assert typeDescriptor.getChildren() != null:"Catalog Type";
```
[the remark this answers is not in the document]
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/iapi/types/DataTypeDescriptor.java]: Sorry for the careless copy-paste error. For ArrayType, its underneath child type should be set. I've corrected the error message. Also typeDescriptor should not be null.

**Comment by yxia92** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**